### PR TITLE
feat: add time frequencies kwtts usage charts.

### PR DESCRIPTION
**Pull request by ceoaliongroo**
#265 #219 

workserror

### CODE REVIEW

**Comment by ceoaliongroo** [comment]
__Update__:

screen shot 2015-01-25 at 4 13 29 pm

**Comment by ceoaliongroo** [comment]
@shahardo, you can used.

**Comment by ceoaliongroo** [comment]
After merge with master found error in ``ChartUsage`` service have dependencies in ``$stateParms`` properties  that is better handle in the controller.
https://github.com/Gizra/negawatt-server/blob/master/client/app/scripts/services/chartUsage.js#L148-L166

Also the lazy load is handle without the resolve from the ui-router
https://github.com/Gizra/negawatt-server/blob/master/client/app/scripts/controllers/usage.js#L12-L16

The object of the chart is replaced every time, and not updated their properties, this return error in the ``google-chart-angular`` module and missing default configuration.

I open a issue to improve and resolve the issue, from this branch.  
https://github.com/Gizra/negawatt-server/blob/265/client/app/scripts/controllers/usage.js#L30
Related #365

File: behat/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -261,11 +283,40 @@ private function waitForXpathNode($xpath, $appear = TRUE) {
...
    * @throws Exception
    */
-  private function waitForTextNgElement($csspath, $text, $appear = TRUE) {
-    $this->waitFor(function($context) use ($csspath, $text, $appear) {
+  private function waitForTextNgElement($csspath, $text) {
+    $this->waitFor(function($context) use ($csspath, $text) {
       try {
-        $element_text = $context->getSession()->evaluateScript('angular.element("' + $csspath + '").text();');
+        $element_text = $context->getSession()->evaluateScript('angular.element("' . $csspath . '").text();');
+        print_r($element_text);
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: remove

File: client/app/scripts/controllers/usage.js [hunk 2]
```diff
@@ -8,12 +8,42 @@
...
+        $stateParams.chartFreq = this.frequencies[this.$index].type;
+        // Load electricity data in the chart according the chart frequency.
+        $scope.isLoading = true;
+
+        ChartUsage.get(account.id, $stateParams).then(function(data) {
+          $scope.usageChartData = data;
+          $scope.isLoading = false;
+        });
+
+        //console.log('select Tab validated');
```
Comment (amitaibu) [inline client/app/scripts/controllers/usage.js]: remove.

File: client/app/scripts/controllers/usage.js [hunk 3]
```diff
@@ -8,34 +8,69 @@
...
  */
 angular.module('negawattClientApp')
-  .controller('UsageCtrl', function ($scope, $stateParams, account, usage, meters, ChartUsage) {
+  .controller('UsageCtrl', function ($scope, $q, $location, $stateParams, account, usage, meters, ChartUsage) {
     // Get data from the cache, since 'usage' might not be up to date
     // after lazy-load.
-    ChartUsage.get(account.id, $stateParams).then(function(data) {
-      $scope.usageChart = data;
+    $scope.frequencies = ChartUsage.getFrequencies();
+    $scope.cssStyle = 'height:210px; width:500px;';
```
Comment (amitaibu) [inline client/app/scripts/controllers/usage.js]: remove - move to class

File: behat/features/bootstrap/FeatureContext.php [hunk 4]
```diff
@@ -112,21 +112,21 @@ public function iShouldSeeMarkers($markers, $equals = TRUE) {
...
     // check if exist and is selected.
-    $this->waitFor(function($context) use ($selected_src_image, $appear) {
+    $this->waitFor(function($context) use ($selected_src_image) {
       try {
         // Get an array of string <img src="...">, filled with the value of the src attribute of the marker icon image.
-        $marker_attr_src = $context->getSession()->evaluateScript('angular.element(".leaflet-marker-icon").map(function(index, element){ return angular.element(element).attr("src") });');
-        if (in_array($selected_src_image, $marker_attr_src)) {
-          return $appear;
+        $marker_selected = $context->getSession()->evaluateScript('angular.element(".leaflet-marker-icon").map(function(index, element){ return angular.element(element).attr("src") }).toArray().indexOf("' . $selected_src_image . '");');
+        if ($marker_selected !== -1) {
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: ``return (bool)$marker_selected;``

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**